src/main/kotlin/se/uulm/snowballr/backend/model/SnowballRException.kt

```diff
@@ -23,13 +23,25 @@ sealed class SnowballRException(
      * @param entityType The type of the entity that could not be found.
      * @param entityIds The missing entity's ID(s).
      * @param identifierType The type of the entity's identifier. Defaults to [IdentifierType.ID].
+     * @param location The location where the entity could not be found, e.g., the project or paper. Should start with " in ...".
      */
     open class NotFoundException(
         entityType: EntityType,
         vararg entityIds: String,
         identifierType: IdentifierType = IdentifierType.ID,
+        location: String = "",
     ) : SnowballRException(
-        "${entityType.singularUpper()} ${displayEntityIds(entityIds.toList(), identifierType)} not found.",
+        "${entityType.singularUpper()} ${displayEntityIds(entityIds.toList(), identifierType)} not found $location.",
```

Comment:
```suggestion
        "${entityType.singularUpper()} ${displayEntityIds(entityIds.toList(), identifierType)} not found$location.",
```
Otherwise you would have a space between "found" and "." if you don't provide any location and the empty string is used.